sink/datadog.go

```diff
@@ -470,6 +504,41 @@ func (s *Datadog) estimateSize(metrics []datadogV2.MetricSeries) (int, error) {
 	return size / len(metrics), nil
 }
 
+func (s *Datadog) uniqIdentifierForMetric(name string, tags []string) string {
+	// sort the list of tags to guarantee consistent ordering
+	sort.Strings(tags)
+
+	var key string
+	key += name
+	for _, t := range tags {
+		v := s.intForTag(t)
+		key += "_" + strconv.Itoa(int(v))
+	}
+	return key
+}
+
+// intForTag returns an uint16 for each tag so the uniq identifier for each metric and tags
+// combination we need to keep track of previous counter values are short.
+// it also maintains a separate counter for each tag key so the numbers don't get large
+func (s *Datadog) intForTag(tag string) uint16 {
+	v, ok := s.tagToInt[tag]
+	if !ok {
+		tokens := strings.Split(tag, ":")
+		next, ok := s.tagValues[tokens[0]]
```

Comment:
I'm realizing that if we don't also include the _value_ for the tag someplace that this won't work properly, as we are creating a "unique " identifier on only the tag names and not the values.

Reply:
You are right, as for metrics with tablenames (or db) as tags would need them in the token to uniquely identify them.

We can just use the full `tag:value` pair rather than splitting it.